src/Models/BaseElement.php

```diff
@@ -300,15 +300,18 @@ public function getCMSFields()
                 $fields->removeByName('Style');
             }
 
-            // Support for new history viewer in SS 4.2+
-            if (class_exists(HistoryViewerField::class)) {
-                Requirements::javascript('dnadesign/silverstripe-elemental:client/dist/js/bundle.js');
-
-                $historyViewer = HistoryViewerField::create('ElementHistory');
-                $fields->addFieldToTab('Root.History', $historyViewer);
-
-                $fields->fieldByName('Root.History')
-                    ->addExtraClass('elemental-block__history-tab tab--history-viewer');
+            // Check object exists before pushing History Tab
+            if($this->ID) {
```

Comment:
Can you use `$this->isInDB()` here? It's pretty similar logic but the method might be updated to more accurately determine if the object is persisted.

Reply:
Also, combining the conditions is nicer because then you're only updated one line instead of 12.